### PR TITLE
Add a fine grain FireFox error handler

### PR DESCRIPTION
**Pull request by renajohn**
FireFix extension is not handling errors, reporting the same
generic error for any load errors (connection error, ssl, redirect
loop, ...). This PR adds a fine grained error handling and introduce
a new error type, redirect loop (13000). This error is also detected
in Chrome. IE does report it.

### CODE REVIEW

File: agent/browser/firefox/extension/chrome/content/overlay.js [hunk 1]
```diff
@@ -346,6 +378,8 @@ var wptExtension = {
     // Use the load event on the global browser object to see when the
     // page gets the onload event.
     if (!g_initialized) {
+      wpt.moz.logInfo('Init');
```
Comment (ocr) [inline agent/browser/firefox/extension/chrome/content/overlay.js]: leftover?

Comment (renajohn) [inline agent/browser/firefox/extension/chrome/content/overlay.js]: Yep... :)

File: agent/browser/firefox/extension/chrome/content/overlay.js [hunk 2]
```diff
@@ -55,6 +55,8 @@ var g_processing_task = false;
 var g_started = false;
 var g_initialized = false;
 var g_webdriver_mode = false;
+var g_lastError = -1;
```
Comment (ocr) [inline agent/browser/firefox/extension/chrome/content/overlay.js]: I don't get why this global variable is needed?

**Comment by ocr** [comment]
:+1: :+1:

**Pull request merged**